llama_index/response_synthesizers/base.py

```diff
@@ -99,16 +101,17 @@ def _prepare_response_output(
                 source_nodes=source_nodes,
                 metadata=response_metadata,
             )
-        elif response_str is None or isinstance(response_str, Generator):
+        if response_str is None or isinstance(response_str, Generator):
             return StreamingResponse(
                 response_str,
                 source_nodes=source_nodes,
                 metadata=response_metadata,
             )
-        else:
-            raise ValueError(
-                f"Response must be a string or a generator. Found {type(response_str)}"
-            )
+        if response_str is not None and response_str.__class__.__name__ != 'str':
```

Comment:
This is pretty hacky. I know we spent a while debugging this on discord, but there must be an easier way 🤔 

I think if the base synthesizer held the output_cls instead of the LLMPredictor, the instance check here would be easy